### PR TITLE
ko: init at 0.8.3

### PR DESCRIPTION
**Pull request by NickCao**
###### Motivation for this change
ko: init at 0.8.3

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ko/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+buildGoModule rec {
+  pname = "ko";
+  version = "0.8.3";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    leaveDotGit = true;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ko/default.nix:14]:
```suggestion
```

Comment (NickCao) [reply pkgs/development/tools/ko/default.nix:14]: This is required for running tests.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130145` at 1cd629d8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/836f0c948dae96d880028a83d22de873">ko</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/tools/ko/default.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/development/tools/ko/default.nix:14]: leaveDotGit ssems to be causing some indeterminism in fetchFromGitHub, I'll find some workaround.

Comment (SuperSandro2000) [reply pkgs/development/tools/ko/default.nix:14]: > This is required for running tests.

Can we patch the tests instead or just skip the ones relying on it?

Comment (NickCao) [reply pkgs/development/tools/ko/default.nix]: running `git init` in preCheck works.

File: pkgs/development/tools/ko/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-LoOXZY4uF7GSS3Dh/ozCsLJTxgmPmZZuEisJ4ShjCBc=";
+  };
+
+  vendorSha256 = null;
+  excludedPackages = "test";
+  nativeBuildInputs = [ git ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/ko/default.nix:20]:
```suggestion
  checkInputs = [ git ];
```

**Pull request merged**